S2VX.Game/Editor/ToolState/SelectToolState.cs

```diff
@@ -102,7 +102,7 @@ public class SelectToolState : S2VXToolState {
                 foreach (var noteAndTime in SelectedNoteToTime) {
                     var note = noteAndTime.Key;
                     var newTime = GetClosestTickTime(gameTimeAtMouse) + NoteToDragPointDelta[note];
-                    note.EndTime = newTime;
+                    note.UpdateEndTime(newTime);
```

Comment:
Here's basically the magic. To update both the Approach and Note together, you call a corresponding Update function inside of Note like this.